docs/eventlisteners.md

```diff
@@ -260,7 +256,7 @@ spec:
     - name: cel-trig
       interceptors:
         - cel:
-            filter: "headers.match('X-GitHub-Event', 'pull_request')"
+            filter: "headers.matches('X-GitHub-Event', 'pull_request')"
```

Comment:
I realised this after I hit /approve...
I think we added a custom `match` function : https://github.com/tektoncd/triggers/blob/master/pkg/interceptors/cel/cel.go

Reply:
I'm not very CEL savvy. Why did we create a custom `match` function?

Reply:
Some discussion in the original PR here: https://github.com/tektoncd/triggers/pull/285#discussion_r360061744

Reply:
Perhaps worth clarifying this in the docs? The built in `matches` vs `match` can get confusing...
@bigkevmcd

Reply:
If I understand correctly, it's used specifically to match agains a header with one value? Doesn't `headers.X-GitHub-Event[0].matches('pull_request')` have the some functionality (I'm probably missing something 😅 ).

Assuming I'm missing something, could we rename the built-in CEL function from `match` to `headerMatch` or `singleHeaderMatch` or something to be more explicit? I assumed it was a typo for `matches`.

Reply:
Fair point...what do you think @bigkevmcd ?

Reply:
A couple of possibilities come to mind, I could add an overlay function  for requests something like `canonicalHeader` which would lookup the header using the Go http.Header `.Get` and return it.

This _might_ fall foul of the change from String to []byte for functions, I'd need to check.

Either way, yes, happy to rename `match` to `headerMatch`.

Reply:
Actually, looking at this again, the reason I called it match is that the expression reads as `header.match`, duplicating `match` again with `header.matchHeader` feels wrong.

Reply:
But we don't want people doing `body.match`, right? Does the function match the first element in the array? It could be `header.matchFirst` or `header.matchFirstElement`?